library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Types/CString.hs

```diff
@@ -0,0 +1,41 @@
+-- | Defines a type for \00 terminated strings
+module Juvix.Backends.LLVM.Codegen.Types.CString (
+  CString
+  , fromText
+  , toText
+  , length
+  , encodeUtf8
+  )
+
+where
+
+import           Data.Hashable
+import qualified Data.Text     as Text
+import           Juvix.Library hiding (length, encodeUtf8)
+import qualified Juvix.Library as J
+import qualified Prelude
+
+newtype CString = CString Text
+  deriving (Eq, Ord, Semigroup, Monoid, Hashable)
+
+instance Show CString where
+  show = Prelude.show . toText
+
+instance ConvertText CString Text where
+  toS = toText
+
+instance ConvertText Text CString where
+  toS = fromText
+
+fromText :: Text -> CString
+fromText = CString
+
+toText :: CString -> Text
+toText (CString t) = t <> "\00"
```

Comment:
This doesn't seem to pass a roundtrip test

Reply:
Indeed.
I think that the best solution is to eliminate the instance `ConvertText CString Text` and don't export `toText`, as it is not used outside of the module and it makes more sense to extract the utf8 encoded `ByteString` rather than `Text`. Do you agree?

Reply:
Yes, sounds good